jupyter_server/services/sessions/handlers.py

```diff
@@ -161,6 +161,8 @@ async def patch(self, session_id):
                 changes["kernel_id"] = kernel_id
             elif model["kernel"].get("name") is not None:
                 kernel_name = model["kernel"]["name"]
+
+            try:
                 kernel_id = await sm.start_kernel_for_session(
                     session_id,
                     kernel_name=kernel_name,
```

Comment:
Some of the test failures are relavant:

```
  File "D:\a\jupyter_server\jupyter_server\jupyter_server\services\sessions\handlers.py", line 168, in patch
    kernel_name=kernel_name,
UnboundLocalError: local variable 'kernel_name' referenced before assignment
```

I think the `try:` needs one more indentation level

Reply:
Oh sorry, yes. I'm tired today.